src/main/java/baritone/command/defaults/FarmCommand.java

```diff
@@ -34,8 +43,50 @@ public FarmCommand(IBaritone baritone) {
 
     @Override
     public void execute(String label, IArgConsumer args) throws CommandException {
-        args.requireMax(0);
-        baritone.getFarmProcess().farm();
+        args.requireMax(2);
+        int range = 0;
+        BetterBlockPos origin;
+        //range
+        if (args.has(1)) {
+            range = args.getAs(Integer.class);
+        }
+        //waypoint
+        if (args.hasExactly(1)){
+            IWaypoint[] waypoints = args.getDatatypeFor(ForWaypoints.INSTANCE);
+            IWaypoint waypoint = null;
+            if (args.hasAny() && args.peekString().equals("@")) {
+                long timestamp = args.getAs(Long.class);
+                for (IWaypoint iWaypoint : waypoints) {
+                    if (iWaypoint.getCreationTimestamp() == timestamp) {
+                        waypoint = iWaypoint;
+                        break;
+                    }
+                }
+                if (waypoint == null) {
+                    throw new CommandInvalidStateException("Timestamp was specified but no waypoint was found");
+                }
+            } else {
+                switch (waypoints.length) {
+                    case 0:
+                        throw new CommandInvalidStateException("No waypoints found");
+                    case 1:
+                        waypoint = waypoints[0];
+                        break;
+                    default:
+                        break;
+                }
+            }
+            if (waypoint == null) {
+                throw new CommandInvalidStateException("Multiple waypoints were found");
```

Comment:
Move this throw up into the `default:`

Reply:
How do you mean?

Reply:
He means you should put the `throw new CommandInvalidStateException("Multiple waypoints were found");` into the `default:` block and remove the if

Reply:
I'm unsure of what you mean.

Reply:
they mean this
`default:
 throw new CommandInvalidStateException("Multiple waypoints were found");     
break;`

Reply:
Sorry, was not refreshing the page and didn't see any replays. I thought you meant the interface default.